inference-engine/src/transformations/src/transformations/smart_reshape/reshape_with_hc_output.cpp

```diff
@@ -0,0 +1,83 @@
+// Copyright (C) 2018-2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/smart_reshape/reshape_with_hc_output.hpp"
+
+#include <memory>
+#include <vector>
+
+#include <ngraph/ngraph.hpp>
+#include <ngraph/pattern/matcher.hpp>
+#include <ngraph/rt_info.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <ngraph/opsets/opset4.hpp>
+
+void relax_hc_reshape_followed_by_matmul(std::shared_ptr<ngraph::opset4::MatMul> matmul, std::shared_ptr<ngraph::Node> reshape,
+                                         ngraph::Output<ngraph::Node> shape_source, std::shared_ptr<ngraph::opset4::Constant> reshape_pattern,
+                                         const bool & reshape_is_A_input) {
```

Comment:
Please do not pass a const bool by reference. Explanation why: https://stackoverflow.com/questions/11932614/in-c-is-it-bad-to-pass-a-const-bool-by-reference
```suggestion
                                        bool reshape_is_A_input) {
```